gc/base/standard/Scavenger.cpp

```diff
@@ -3072,25 +3078,27 @@ MM_Scavenger::abandonSurvivorTLHRemainder(MM_EnvironmentStandard *env)
 }
 
 void
-MM_Scavenger::abandonTenureTLHRemainder(MM_EnvironmentStandard *env)
+MM_Scavenger::abandonTenureTLHRemainder(MM_EnvironmentStandard *env, bool preserveRemainders)
 {
 	if (NULL != env->_tenureTLHRemainderBase) {
 		Assert_MM_true(NULL != env->_tenureTLHRemainderTop);
-		env->_scavengerStats._tenureDiscardBytes += (uintptr_t)env->_tenureTLHRemainderTop - (uintptr_t)env->_tenureTLHRemainderBase;
 		_tenureMemorySubSpace->abandonHeapChunk(env->_tenureTLHRemainderBase, env->_tenureTLHRemainderTop);
-		env->_tenureTLHRemainderBase = NULL;
-		env->_tenureTLHRemainderTop = NULL;
+
+		if (!preserveRemainders){
+			env->_scavengerStats._tenureDiscardBytes += (uintptr_t)env->_tenureTLHRemainderTop - (uintptr_t)env->_tenureTLHRemainderBase;
+			env->_tenureTLHRemainderBase = NULL;
+			env->_tenureTLHRemainderTop = NULL;
+		}else if(env->isMasterThread()){
+			/* Save the remainder references to extension, to be reloaded during masterSetupForGC() */
+			_extensions->_tenureTLHRemainderTop = env->_tenureTLHRemainderTop;
```

Comment:
add in the comment, this is need for master thread since it is an implicit thread and in general change from cycle to cycle